### PR TITLE
Feature/invocation

### PR DESCRIPTION
**Pull request by priyavadan**
use site config to enable/disable origen invocation tracking as well as set the invocation  url

### CODE REVIEW

**Comment by priyavadan** [comment]
tests are passing in the local environment, not sure what is wrong with Github Actions

image

**Comment by pderouen** [comment]
https://github.com/Origen-SDK/origen/blob/b232538178b80e566e78d4e74d8c5c6807b800ec/spec/callbacks_spec.rb#L148-L150

Looks like that failing test has a history of behaving differently between local and CI runs.

**Comment by pderouen** [comment]
It fails locally for me the same as with CI

image

**Comment by priyavadan** [comment]
Interesting, none of the changes affect callback testing though.

Wonder why it fails locally for you but not for me...

**Comment by pderouen** [comment]
I'm poking around to see if I can figure out why. But, <code>program_generated</code> is not getting called.

File: lib/origen/application/runner.rb [hunk 1]
```diff
@@ -134,30 +134,40 @@ def prepare_and_validate_workspace(options = {})
...
-        #  # that threads can't be trusted not to block
-        #  unless Origen.running_remotely? # || Origen.running_on_windows?
-        #    record_invocation = Thread.new do
-        #      Origen.client.record_invocation(options[:action]) if options[:action]
-        #    end
-        #  end
-        # rescue
-        #  # Don't allow this to kill an origen command
-        # end
+        if Origen.site_config.record_invocation == true
```
Comment (ginty) [inline lib/origen/application/runner.rb:137]: I assume this just returns `nil` today for any attribute not mentioned in the config?

Comment (priyavadan) [reply lib/origen/application/runner.rb:137]: correct!

**Review by ginty — APPROVED** [review]
LGTM

**Review by pderouen — APPROVED** [review]

**Pull request merged**